notebook/notebookapp.py

```diff
@@ -1820,10 +1854,17 @@ def launch_browser(self):
         else:
             open_file = self.browser_open_file
 
-        b = lambda: browser.open(
-            urljoin('file:', pathname2url(open_file)),
-            new=self.webbrowser_open_new)
-        threading.Thread(target=b).start()
+        if not self.use_redirect_file:
+            if browser:
```

Comment:
You don't need to check `if browser:` here - it's already checked for an early return at the start of the function.

Reply:
This makes sense now. I was still pretty unfamiliar with the code base when I created this and was making some assumptions.